qemu/tests/numa_memdev_options.py

```diff
@@ -0,0 +1,110 @@
+import logging
+
+from avocado.utils import process
+
+from virttest import error_context
+from virttest import utils_misc
+
+
+def check_host_numa_node_amount(test, params):
+    """
+    Check host NUMA node amount
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    """
+    host_numa_nodes = int(process.system_output("%s" % params['numa_hardware_cmd']).split()[1])
+    if host_numa_nodes < 2:
+        test.cancel("The host numa nodes should be at least 2! But there is %d." % host_numa_nodes)
+
+
+def check_query_memdev(test, params, vm):
+    """
+    Check memory info in query-memdev
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param vm: VM object
+    """
+    mem_devs = params['mem_devs'].split()
+    query_list = vm.monitor.info("memdev")
+    if len(mem_devs) != len(query_list):
+        test.fail("%d memory devices in query-memdev, but not %d!"
+                  % (len(query_list), len(mem_devs)))
+    policy = params['policy_mem']
+    for dev in query_list:
+        mem_dev = dev['id'].split('-')[1]
+        if dev['policy'] != policy:
+            test.fail("memdev = %s: 'policy' is '%s', but not '%s'!"
+                      % (mem_dev, dev['policy'], policy))
+        prealloc = (params['prealloc_%s' % mem_dev] == 'yes')
+        if dev['prealloc'] != prealloc:
+            test.fail("memdev = %s: 'prealloc' is not '%s'!"
+                      % (mem_dev, params['prealloc_' % mem_dev]))
+        if policy == 'default':
+            continue
+        else:
+            host_node = str(dev['host-nodes'][0])
+            if host_node != params['host-nodes_%s' % mem_dev]:
+                test.fail("memdev = %s: 'host-nodes' is '%s', but not '%s'!"
+                          % (mem_dev, host_node, params["host-nodes_%s" % mem_dev]))
+
+
+def check_memory_in_procfs(test, params, vm):
+    """
+    Check memory info in procfs
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param vm: VM object
+    """
+    qemu_pid = vm.get_pid()
+    policy = params['policy_mem']
+    if policy == 'preferred':
+        policy = 'prefer'
+    mem_path = params.get("mem-path", None)
+    for mem_dev in params['mem_devs'].split():
+        mem_size = int(float(utils_misc.normalize_data_size(params['size_%s' % mem_dev], "K")))
+        smaps = process.system_output("grep -1 %d /proc/%d/smaps" % (mem_size, qemu_pid))
+        if mem_path and (mem_path not in smaps):
+            test.fail("memdev = %s: mem-path '%s' is not in smaps '%s'!"
+                      % (mem_dev, mem_path, smaps))
+        mem_start = smaps.split('-')[0]
+        numa_maps = process.system_output("grep %s /proc/%d/numa_maps" % (mem_start, qemu_pid))
```

Comment:
`read_from_numa_maps()` can be used that can reduce some code on creating path.

Reply:
What I need is the second part "policy[:host-nodes]" of specific memory address in numa_maps, while read_from_numa_maps(pid, key) has ability to get value of specific key like "anon", "dirty", ..., in numa_maps(contains all memory address in numa_maps). So, IMHO, it's not appropriate to use read_from_numa_maps() in my case. Thanks!